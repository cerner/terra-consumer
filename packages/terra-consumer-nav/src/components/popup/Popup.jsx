import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Grid from 'terra-grid';
import IconClose from 'terra-icon/lib/icon/IconClose';
import TerraPopup from 'terra-popup';
import styles from './Popup.scss';

const cx = classNames.bind(styles);


const propTypes = {
  /**
   * Title to be rendered on top of the popup content
   */
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
  ]),
  /**
   * Boolean value to hide/display popup header with title and close button
   */
  hasHeader: PropTypes.bool,
  /**
   * Content to be displayed in a dialog
   */
  popupContent: PropTypes.element,
  /**
   * Boolean value to render the popup
   */
  isOpen: PropTypes.bool,
  /**
   * Required callback function for use by parent component to update state.
   */
  closePopup: PropTypes.func,
};

const defaultProps = {
  title: null,
  hasHeader: false,
  popupContent: null,
  isOpen: false,
  closePopup: null,
};

const Popup = ({
 title,
 hasHeader,
 popupContent,
 isOpen,
 closePopup,
 ...customProps
}) => {
  const popupHeader = (
    <Grid className={cx('popup-header')}>
      <Grid.Row>
        <Grid.Column col={10}>
          <div className={cx('popup-title')}>{title}</div>
        </Grid.Column>
        <Grid.Column className={cx('close-button-alignment')} col={2}>
          <button className={cx('close-button')} onClick={() => closePopup()}><IconClose className={cx('close-icon')} /></button>
        </Grid.Column>
      </Grid.Row>
    </Grid>);

  return (
    <TerraPopup
      {...customProps}
      isOpen={isOpen}
      onRequestClose={closePopup}
      isHeaderDisabled
    >
      <div>
        {hasHeader && popupHeader}
        {popupContent}
      </div>
    </TerraPopup>
  );
};

Popup.propTypes = propTypes;
Popup.defaultProps = defaultProps;

export default Popup;
