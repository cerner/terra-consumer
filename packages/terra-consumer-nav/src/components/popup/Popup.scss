:local {
  .popup-header {
    background: var(--terra-consumer--popup-header-background, #64696c);
    display: inline-block;
    padding: 5px 0;
    text-align: left;
    width: 100%;
  }

  .popup-title {
    color: var(--terra-consumer--popup-title-color, #fff);
    font-size: 20px;
    line-height: 30px;
    padding-left: 15px;
  }

  .close-icon {
    height: 16px;
    width: 16px;
  }

  .close-button {
    background: none;
    border: 0;
    color: var(--terra-consumer--popup-close-button-color, #fff);
    cursor: pointer;
    font-size: 16px;
    margin: 0;
    padding: 0;
    position: absolute;
    right: 15px;
    top: 10px;
  }

}
