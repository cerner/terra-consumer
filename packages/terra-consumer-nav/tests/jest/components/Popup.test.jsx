import React from 'react';
import Popup from '../../../src/components/popup/Popup';


const popupContent = <div> This is a test content</div>;
const mockFn = jest.fn();

describe('Popup', () => {
  // Snapshot Tests
  it('should render a pop with title,close button and content', () => {
    const wrapper = shallow(<Popup
      title="Title"
      hasHeader
      targetRef={mockFn}
      closePopup={mockFn}
      contentWidth="320"
      contentHeight="240"
      contentAttachment="middle center"
      popupContent={popupContent}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a popup without header(title & close button)', () => {
    const wrapper = shallow(<Popup
      title="Title"
      hasHeader={false}
      targetRef={mockFn}
      closePopup={mockFn}
      contentWidth="320"
      contentHeight="240"
      contentAttachment="middle center"
      popupContent={popupContent}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});

