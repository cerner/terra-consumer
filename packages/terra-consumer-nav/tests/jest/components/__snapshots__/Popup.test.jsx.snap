// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render a pop with title,close button and content 1`] = `
<Popup
  boundingRef={null}
  classNameArrow={null}
  classNameContent={null}
  classNameOverlay={null}
  contentAttachment="middle center"
  contentHeight="240"
  contentWidth="320"
  isArrowDisplayed={false}
  isHeaderDisabled={true}
  isOpen={false}
  onRequestClose={[Function]}
  targetRef={[Function]}
>
  <div>
    <div
      className="popup-header"
    >
      <p
        className="popup-title"
      >
        Title
        <button
          className="close-button"
          onClick={[Function]}
        >
          <IconClose
            className="close-icon"
            data-name="Layer 1"
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
      </p>
    </div>
    <div>
       This is a test content
    </div>
  </div>
</Popup>
`;

exports[`Popup should render a popup without header(title & close button) 1`] = `
<Popup
  boundingRef={null}
  classNameArrow={null}
  classNameContent={null}
  classNameOverlay={null}
  contentAttachment="middle center"
  contentHeight="240"
  contentWidth="320"
  isArrowDisplayed={false}
  isHeaderDisabled={true}
  isOpen={false}
  onRequestClose={[Function]}
  targetRef={[Function]}
>
  <div>
    <div>
       This is a test content
    </div>
  </div>
</Popup>
`;
